### PR TITLE
Separate TcpKeepAliveIdle and TcpKeepAliveInterval check logic

### PR DESCRIPTION
**Pull request by ValdikSS**
This patch allows to separately configure both parameters.

### CODE REVIEW

**Review by kslr — APPROVED** [review]

**Review by xiaokangwang — APPROVED** [review]

File: transport/internet/sockopt_darwin.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ const (
 	TCP_FASTOPEN_SERVER = 0x01 // nolint: revive,stylecheck
 	// TCP_FASTOPEN_CLIENT is the value to enable TCP fast open on darwin for client connections.
 	TCP_FASTOPEN_CLIENT = 0x02 // nolint: revive,stylecheck
+	// syscall.TCP_KEEPINTVL is missing on some darwin architectures.
+	sysTCP_KEEPINTVL = 0x101 // nolint: revive,stylecheck
```
Comment (database64128) [inline transport/internet/sockopt_darwin.go:13]: Which darwin architectures are missing `TCP_KEEPINTVL`? We are using `x/sys/unix` here, not `syscall`, please double check.

File: transport/internet/sockopt_darwin.go [hunk 2]
```diff
@@ -24,9 +26,16 @@ func applyOutboundSocketOptions(network string, address string, fd uintptr, conf
...
+		if config.TcpKeepAliveIdle > 0 || config.TcpKeepAliveInterval > 0 {
+			if config.TcpKeepAliveIdle > 0 {
+				if err := unix.SetsockoptInt(int(fd), unix.IPPROTO_TCP, unix.TCP_KEEPALIVE, int(config.TcpKeepAliveInterval)); err != nil {
+					return newError("failed to set TCP_KEEPINTVL", err)
+				}
+			}
+			if config.TcpKeepAliveInterval > 0 {
+				if err := unix.SetsockoptInt(int(fd), unix.IPPROTO_TCP, sysTCP_KEEPINTVL, int(config.TcpKeepAliveIdle)); err != nil {
+					return newError("failed to set TCP_KEEPIDLE", err)
+				}
```
Comment (database64128) [inline transport/internet/sockopt_darwin.go:38]: I think moving `setsockopt(TCP_KEEPALIVE)` and `setsockopt(TCP_KEEPINTVL)` out of the `if config.TcpKeepAliveIdle > 0 || config.TcpKeepAliveInterval > 0` block would make it more readable.

**Review by database64128 — CHANGES_REQUESTED** [review]

File: transport/internet/sockopt_darwin.go (hunk 1 above)
Comment (ValdikSS) [reply transport/internet/sockopt_darwin.go:13]: I've copied this from https://github.com/golang/go/blob/master/src/net/tcpsockopt_darwin.go

Comment (database64128) [reply transport/internet/sockopt_darwin.go:13]: That's not where you should be looking at.

https://github.com/golang/sys/blob/1d35b9e2eb4edf581781c7f3e2a36fac701f0a24/unix/zerrors_darwin_amd64.go#L1459

https://github.com/golang/sys/blob/1d35b9e2eb4edf581781c7f3e2a36fac701f0a24/unix/zerrors_darwin_arm64.go#L1459

`TCP_KEEPINTVL` is present for both `darwin-amd64` and `darwin-arm64`. You can safely use `unix.TCP_KEEPINTVL`.

File: transport/internet/sockopt_darwin.go (hunk 2 above)
Comment (xiaokangwang) [reply transport/internet/sockopt_darwin.go]: Your recommendation have been adapted at 406e0f7ac6614e53870687725dfc8e48a142be95.

File: transport/internet/sockopt_darwin.go (hunk 1 above)
Comment (xiaokangwang) [reply transport/internet/sockopt_darwin.go]: Your recommendation have been adapted at 406e0f7ac6614e53870687725dfc8e48a142be95.

**Review by database64128 — APPROVED** [review]

**Comment by ValdikSS** [comment]
@xiaokangwang, thanks for the fix. I'll adapt it for #1483 and update shortly.

**Pull request merged**